### DISCUSSION TITLE
RP2040 W wireless pico

### DISCUSSION THREAD

**Discussion by MYCAMEL222** [earlephilhower/arduino-pico#665] [General]
Just saw this this morning. Can't wait to give it a try.

https://datasheets.raspberrypi.com/picow/pico-w-datasheet.pdf

"Pico W contains an on-board 2.4GHz wireless interface using the Infineon CYW43439, which has the following features:
• WiFi 4 (802.11n), Single-band (2.4 GHz)
• WPA3
• SoftAP (Up to 4 clients)
The antenna is an onboard antenna licensed from ABRACON (formerly ProAnt). The wireless interface is connected via
SPI to the RP2040."

**Comment by earlephilhower** [comment]
One one hand, that's awesome.  Networking was the only thing missing from the Pico.

On the other hand, darn it.  More code to write.  😆

**Reply by skibum-za** [reply]
Just waiting for mine to arrive tomorrow. Might have to play using python for a while, although very much hoping support for this can be added. It would be amazing.

**Reply by earlephilhower** [reply]
Also looks like it has BT and BLE!  No availability here in the US, yet, as far as I can see.  We'll see next month how things look.  For the original Pico I was able to get a pack of 5 w/in a week of launch...

**Reply by MYCAMEL222** [reply]
"Eagle-eyed readers of [datasheets](https://datasheets.raspberrypi.com/picow/pico-w-datasheet.pdf) will notice that CYW43439 supports both Bluetooth Classic and Bluetooth Low-Energy: we have not enabled Bluetooth on Pico W at launch, but may do so in the future....."

From https://www.raspberrypi.com/news/raspberry-pi-pico-w-your-6-iot-platform/

**Reply by MYCAMEL222** [reply]
Looking at the R2040 w schematic, it appears the BT_UART pins are not connected to anything. No amount of programming can fix that.

I wonder why they would release the 2040W without BT/BLE..?

**Reply by earlephilhower** [reply]
Seems it has to do w/FCC etc. radio licensing.  Since it's a chip, not a carrier board (like the ESPxxxs), they need to submit it to the relevant country certification authorities.  If they just soldered on a castellated(sp?) board, they would not need this since the carrier board would include the antenna and be cert'd by the original manufacturer.

**Reply by sstaub** [reply]
Ordered also some, there is no API for Arduino, so we have to do it.
There is an update for the SDK but without a real documentation in the moment.

**Reply by Atorque** [reply]
I got mine a few days ago.  Patiently waiting for you to write your code earlephilhower.  I just wanted to tip my hat at you sir.  I would sound like a total fanboy if I mentioned how I really feel about your pico code, so I will simply say thank you.  You rock!

**Reply by MandoRick** [reply]
You made it possible for me to use my pico when I first got it, so I will wait as long as I must to use my pico W.  Thank you philhower

**Reply by earlephilhower** [reply]
Microcenter pulled through today, :partying_face:.  Was able to snag one this AM before they sold out.  It's blinking away now (picow_blink example).  Well see how this weekend looks for some heavy hacking...
